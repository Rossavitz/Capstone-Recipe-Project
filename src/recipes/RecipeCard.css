.card-body {
    border: 2px black solid;
    width: 35%;
    display: inline-block;
    margin:5px;
    padding: 5px;
}

.card-body h3{
    margin-top: 3px;
}

.card-body p{
    margin-top: 3px;
}

.card-body h5{
    text-align: end;
    margin-bottom: 3px;
}

.card-body .link-style{
cursor: pointer;
text-decoration: underline;
}

.card-body .link-style:hover{
    color: salmon
}