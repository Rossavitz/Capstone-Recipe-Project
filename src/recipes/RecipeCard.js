import React, { useContext } from "react";
import { useNavigate } from "react-router-dom";
import { Link } from "react-router-dom";
import "./RecipeCard.css";
import UserContext from "../auth/UserContext";

function RecipeCard({ id, title, description, tag, search }) {
  const { currUser } = useContext(UserContext);
  let navigate = useNavigate();

  //search tag handler
  function handleClick(e) {
    e.preventDefault();
    search(tag);
  }

  //if no current user, click tag redirects to login
  function noCurrUserHandleClick(e) {
    navigate("/login");
  }

  return (
    <div className="card-body">
      <Link className="RecipeCard card" to={`/recipes/id/${id}`}>
        <h3>{title}</h3>
      </Link>
      <p>{description}</p>

      {currUser ? (
        <h5 className="link-style" onClick={handleClick}>
          #{tag}
        </h5>
      ) : (
        <h5 className="link-style" onClick={noCurrUserHandleClick}>
          #{tag}
        </h5>
      )}
    </div>
  );
}

export default RecipeCard;
