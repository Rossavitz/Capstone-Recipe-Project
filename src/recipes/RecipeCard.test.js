import React from "react";
import { render } from "@testing-library/react";
import RecipeCard from "./RecipeCard";
import { UserProvider } from "../testUtils";
import { BrowserRouter } from "react-router-dom";

it("matches snapshot", function () {
  let item = {
    title: "Test Title",
    description: "Test Description",
    tag: "Tag",
  };
  const { asFragment } = render(
    <BrowserRouter>
      <UserProvider>
        <RecipeCard item={item} />
      </UserProvider>
    </BrowserRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
