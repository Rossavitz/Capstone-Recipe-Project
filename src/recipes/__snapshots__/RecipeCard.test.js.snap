// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="card-body"
  >
    <a
      class="RecipeCard card"
      href="/recipes/id/undefined"
    >
      <h4 />
      <h5 />
      <p />
    </a>
  </div>
</DocumentFragment>
`;
